packages/pyright-internal/src/tests/importStatementUtils.test.ts

```diff
@@ -205,9 +225,24 @@ test('getTextEditsForAutoImportInsertions - multiple modules with existing impor
 //// import os[|/*marker1*/|][|{|"r":"!n!from sys import path as p"|}|][|{|"r":"!n!!n!import numpy"|}|][|{|"r":"!n!!n!from test import join"|}|]
     `;
 
-    const module1 = { moduleName: 'sys', importType: ImportType.BuiltIn, isLocalTypingsFile: false };
-    const module2 = { moduleName: 'numpy', importType: ImportType.ThirdParty, isLocalTypingsFile: false };
-    const module3 = { moduleName: 'test', importType: ImportType.Local, isLocalTypingsFile: false };
+    const module1 = {
+        moduleName: 'sys',
+        importType: ImportType.BuiltIn,
+        isLocalTypingsFile: false,
+        isThirdPartyPyTypedPresent: false,
```

Comment:
Are there any tests for when this should come out as true?

Reply:
Not yet. I'll try to add some. When I created this PR I was feeling unsure about the implementation and didn't want to invest time in tests before I got some feedback.

Reply:
It wasn't clear to me that adding to `importStatementUtils.test.ts` was necessary/interesting. I added a new test that repros the user's scenario in `completions.importInterimFile.fourslash.ts`.

Reply:
I was just thinking for completeness. You added this new logic, but there's no test for the 'true' case. If the 'true' case isn't important, do we even need the flag?

Reply:
I only added `isThirdPartyPyTypedPresent` in this file because it's (now) a required field on `ModuleNameAndType`. I noticed that all of the tests in this file also have `isLocalTypingsFile` set to `false`, I'm assuming for the same reason.

I think the core problem here is that the auto-importer is using `ModuleNameAndType` but doesn't care about `isThirdPartyPyTypedPresent` (or `isLocalTypingsFile`?). I'll try introducing a new type instead.

Reply:
I switched to a new `ModuleImportInfo` type rather than adding `isThirdPartyPyTypedPresent` to `ModuleNameAndType`. I tried removing `isLocalTypingsFile` but it's actually used [here](https://github.com/microsoft/pyrx/blob/4d2c5e6f9da0c3634aff7e6a50bb66dff6706807/packages/pyright/packages/pyright-internal/src/analyzer/importStatementUtils.ts#L742).